File: lib/core/node/models/user_model.js
const mongoose = require('mongoose');

const UserSchema = mongoose.Schema({
    user_id:{
        type: String,
     //   required:true
    },
    user_name:{
        type: String,
        required:true
    },
    user_surname:{
        type: String,
        required:true
    },
    user_email:{
        type: String,
        required:true
    }, 
    user_password:{
        type: String,
        required:true
    },
    user_fotoURL:{
        type: String,
     //   required:true
    },
    user_rooms_id:{
        type: Array,
      //  required:true
    },

}, {timestamps: true });

UserSchema.methods.toJSON = function () {
    const user = this.toObject();
    delete user.createdAt;
    delete user.updatedAt;
    delete user.__v;
    delete user.user_password;

    return user;
}


module.exports = mongoose.model('User', UserSchema);

File: lib/core/models/user_model.dart
//User model

class UserModel {
  List<String>? userRoomsId;
  String? sId;
  String? userName;
  String? userSurname;
  String? userEmail;
  String? userPassword;

  UserModel(
      {this.userRoomsId,
      this.sId,
      this.userName,
      this.userSurname,
      this.userEmail,
      this.userPassword});

  UserModel.fromJson(Map<String, dynamic> json) {
    sId = json['_id'];
    userName = json['user_name'];
    userSurname = json['user_surname'];
    userEmail = json['user_email'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['user_name'] = this.userName;
    data['user_surname'] = this.userSurname;
    data['user_email'] = this.userEmail;
    data['user_password'] = this.userPassword;
    return data;
  }

  Map<String, dynamic> toLoginJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['user_email'] = this.userEmail;
    data['user_password'] = this.userPassword;
    return data;
  }
}
